gui/scripts/syncthing/core/controllers/syncthingController.js

```diff
@@ -106,6 +105,27 @@ angular.module('syncthing.core')
             }
         });
 
+        $scope.$on('HTTPError', function (event, arg) {
+            // Emitted when a HTTP call fails. We use the status code to try
+            // to figure out what's wrong.
+
+            if (navigatingAway || !online) {
+                return;
+            }
+
+            console.log('HTTPError', arg);
+            online = false;
```

Comment:
Don't want to throw up the problem dialog while we are restarting or if it's already up? I didn't think too much about it, just copied the logic from the network problem dialog which is triggered under similar circumstances.